corehq/apps/api/tests.py

```diff
@@ -1879,11 +1875,15 @@ def test_auth(self):
             wrong_domain.name, user_in_wrong_domain_name, user_in_wrong_domain_password
         )
         user_in_wrong_domain.save()
-
-        self.client.login(username=user_in_wrong_domain_name, password=user_in_wrong_domain_password)
-        response = self.client.get(
-            self.single_endpoint(self.report_configuration._id))
-        self.assertEqual(response.status_code, 401)  # 401 is "Unauthorized"
-
-        wrong_domain.delete()
-        user_in_wrong_domain.delete()
+        credentials = base64.b64encode("{}:{}".format(
+            user_in_wrong_domain_name, user_in_wrong_domain_password)
+        )
+        try:
+            response = self.client.get(
+                self.single_endpoint(self.report_configuration._id),
+                HTTP_AUTHORIZATION='Basic ' + credentials
+            )
+            self.assertEqual(response.status_code, 401)  # 401 is "Unauthorized"
+        finally:
+            wrong_domain.delete()
```

Comment:
i think it might be better to `self.addCleanup(wrong_domain.delete)` so that even if the test fails it still cleans up